### ISSUE TITLE
on_success false is broken

### ISSUE THREAD

**Issue by let4be** [hashicorp/nomad#728]
I'm running test golang program inside docker container

I've made 
on_success = false in my restart policy, 
```
restart {
  interval = "5m"
  attempts = 10
  delay = "5s"
  on_success = false
  mode = "delay"
}
```

however I see this in allocation status
```
Recent Events:
Time               Type        Description
09:29:53 01/31/16  Terminated  Exit Code: 0
09:29:22 01/31/16  Started     <none>
09:29:17 01/31/16  Terminated  Exit Code: 0
09:28:47 01/31/16  Started     <none>
09:28:42 01/31/16  Terminated  Exit Code: 0
09:28:12 01/31/16  Started     <none>
09:28:06 01/31/16  Terminated  Exit Code: 0
09:27:36 01/31/16  Started     <none>
```

Nomad v0.2.3

**Comment by let4be** [comment]
It seems like if I change job type from service to batch it works as expected(job isn't being rescheduled)
Is it supposed behavior?

**Comment by dadgar** [comment]
Fixed by #744
